aospy/data_loader.py

```diff
@@ -0,0 +1,416 @@
+                           dtype_in_time=None, intvl_out=None):
+        for name in var.names:
+            try:
+                return self.file_map[intvl_in][name]
+            except KeyError:
+                pass
+        raise KeyError('Files for the var {0} cannot be found in for the '
+                       'intvl_in {1} in this'
+                       ' OneDirDataLoader'.format(var, intvl_in))
+
+
+class GFDLDataLoader(DataLoader):
+    """A data loader that locates files based on GFDL post-processing naming
+    conventions.
+
+    This is an example of a domain-specific custom DataLoader, designed
+    specifically for finding files output by GFDL's model history file
+    post-processing tools.
+
+    Parameters
+    ----------
+    template : GFDLDataLoader
+        Optional argument to specify a base GFDLDataLoader to inherit
+        parameters from
+    data_direc : str
+        Root directory of data files
+    data_dur : int
+        Number of years included per post-processed file
+    data_start_date : datetime.datetime
+        Start date of data files
+    data_end_date : datetime.datetime
+        End date of data files
+
+
+    Examples
+    --------
+    Case without a template to start from.
+
+    >>> base = GFDLDataLoader(data_direc='/archive/control/pp', data_dur=5,
+    ...                       data_start_date=datetime(2000, 1, 1),
+    ...                       data_end_date=datetime(2010, 12, 31))
+
+    Case with a starting template.
+
+    >>> data_loader = GFDLDataLoader(base, data_direc='/archive/2xCO2/pp')
+    """
+    def __init__(self, template=None, data_direc=None, data_dur=None,
+                 data_start_date=None, data_end_date=None):
+        """Create a new GFDLDataLoader"""
+        attrs = ['data_direc', 'data_dur', 'data_start_date', 'data_end_date']
+        if template:
+            for attr in attrs:
+                setattr(self, attr, getattr(template, attr))
+
+            # Override attributes that aren't none
+            if data_direc is not None:
+                self.data_direc = data_direc
+            else:
+                self.data_direc = template.data_direc
+            if data_dur is not None:
+                self.data_dur = data_dur
+            else:
+                self.data_dur = template.data_dur
+            if data_start_date is not None:
+                self.data_start_date = data_start_date
+            else:
+                self.data_start_date = template.data_start_date
+            if data_end_date is not None:
+                self.data_end_date = data_end_date
+            else:
+                self.data_end_date = template.data_end_date
+        else:
+            self.data_direc = data_direc
+            self.data_dur = data_dur
+            self.data_start_date = data_start_date
+            self.data_end_date = data_end_date
+
+    @staticmethod
+    def _maybe_apply_time_shift(da, time_offset=None, **DataAttrs):
```

Comment:
Ha, I see the explanatory text I was referring to doesn't actually exist!  My bad.